app/actions.py

```diff
@@ -258,6 +277,28 @@ def bookmarkRemove(self):
     self.bookmarks = rangeList[:begin] + rangeList[index:]
     return True
 
+  def bookmarkMove(self, upper, delta):
+    """
+    Moves bookmarks up and down.
+    Called after changes to line numbers.
+
+    Args:
+      upper (int): The upper limit on current bookmark positions.
+      delta (int): The number of rows to move.
+
+    Returns:
+      None.
+    """
+    for index in range(len(self.bookmarks)):
```

Comment:
If we think there may be a lot of bookmarks then a bisect may make sense here (if we think there will be a handful of bookmarks at most then it likely doesn't matter much). What do you think about trying a bisect to find the bookmarks affected.

Reply:
I gave that some thought as well and just need a moment to figure out how to put things together.